### ISSUE TITLE
refactor(state): events need their own section of the state tree

### ISSUE THREAD

**Issue by ramfox** [qri-io/frontend#350]
Right now, all events are being stored in the `workflow` section of the state tree. While we definitely need the run info for the current workflow in `workflow` section of the state tree, but we are also mixing that with events that come from other datasets' runs.

We need an `events` section of the state tree & selectors that allow you to select events based on `sessionID`. We can use that `sessionID` to select events based on `runID`, `ref`, `initID` etc. We can also add selectors as needed to filter events further.

The `workflow` reducer can react to events by selecting them from the `events` section of the state tree.

**Comment by ramfox** [comment]
#### moving storing events from `workflow` reducer to new `events` reducer
- [x] create new folder "features/events"
- [x] create new file "eventsState.ts"
- [x] in "eventsState.ts" file, create `eventsReducer`, using the `createReducer` function
- [x] move `RUN_EVENT_LOG` definition from "features/workflow/workflowState.ts" to here
- [x] create `EventState` interface, that as a field `events: EventLogLine[]`, remove the same `events` field in the `WorkflowState`
- [x] move `addRunEvent` function from `workflowState.ts` to the `eventsState.ts`
- [x] move `RUN_EVENTLOG: addRunEvent` from `WorkflowReducer` to `EventsReducer`
- [x] remove references to `workflow.events` field from the `WorkflowReducer`
- [x] move definition `selectRunsFromEventLog` from `workflowState.ts` to `eventState.ts` & rename to `selectRuns`
- [x] remove `selectLatestDryRun` & `selectLatestRun` from `workflowState.ts`
- [x] write `selectRun` that selects a run based on `sessionID` in `eventState.ts` if the session ID is empty, return `undefined`
- [ ] in `workflowState.ts`, write `selectLatestDryRunID` and `selectLatestRunID` that return the `latestDryRun` run id & `latestRun` run id, respectfully
- [ ] replace calls of `selectLatestDryRun` & `selectLatestRun` w/ calls to `selectRun`
- [ ] clean up any imports in any files for functions that have been renamed or moved

The goal is to refactor the state tree, but ensure that all of the current functionality stays the same.

One question to consider, but can be solved at a later time:
- This never "clears" the events, which means we can end up with a VERY long history of events that we probably don't need to hold onto, is there a simple logical way to know when we can "clear" the events?

Once this is refactored, we can adjust what we need to to fix issue #313, which will probably require some more changes to the `workflowState`

**Comment by boandriy** [comment]
Hey, after having a discussion with @ramfox , we have decided to change the way we clear events in two ways:
1. when we are in the workflow and we are editing a code cell that already has an output (meaning that there is an event for it) instead of clearing the whole event list we would use the session id to clear exactly one specific event (meaning that only the event that has output for the cell would be cleared).
2. When the toast appears and user clicks the X button to close it - instead of hiding it in UI and not changing the events state - we would also use the sessionID to clear that specific event that was closed.
